Mathlib/Tactic/Linter/Style.lean

```diff
@@ -68,4 +69,41 @@ initialize addLinter setOptionLinter
 
 end Style.setOption
 
+/-- The `nameCheck` linter emits a warning on declarations whose name is non-standard style.
+(Currently, this only includes declarations whose names include a double underscore.)
+
+**Why is this bad?** Double underscores in theorem names can be considered non-standard style and
+probably have been introduced by accident
+**How to fix this?** Use single underscores to separate parts of a name, following standard naming
+conventions.
+-/
+register_option linter.style.nameCheck : Bool := {
+  defValue := false
+  descr := "enable the `nameCheck` linter"
+}
+
+namespace Style.nameCheck
+
+/-- Checks whether the original input of a given syntax contains "__". -/
+def contains_double_underscore (stx : Syntax) : Bool :=
+  match stx.getSubstring? with
+  | some str => 1 < (str.toString.splitOn "__").length
+  | none => false
+
+@[inherit_doc linter.style.nameCheck]
+def doubleUnderscore: Linter where run := withSetOptionIn fun stx => do
+    unless Linter.getLinterValue linter.style.nameCheck (← getOptions) do
+      return
+    if (← MonadState.get).messages.hasErrors then
+      return
+    for sytax in (← getNames stx) do
+      if contains_double_underscore sytax then
+        Linter.logLint linter.style.nameCheck sytax
+          m!"The declID '{sytax}' contains '__', which does not follow the mathlib naming \
```

Comment:
Can you replace `declID` by `declaration`, please? While we're avoiding technical jargon, let's be consistent.

Reply:
Why is the variable called `sytax` rather than `syntax` or `name`?  `sytax` looks so much like a typo to me!  :smile:

Reply:
it is, will fix

Reply:
done (both the typo and not mentioning `declID`